src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -29,13 +28,14 @@ internal sealed class ObjectWriter
 
         private IFormatterConverter? _formatterConverter;
 
-        internal object[]? _crossAppDomainArray = null;
-        internal List<object>? _internalCrossAppDomainArray = null;
+#pragma warning disable 0649 // Field is never assigned to, and will always have its default value null
+        internal object[]? _crossAppDomainArray;
+#pragma warning restore 0649
```

Comment:
This field can be removed, but pulling on that thread leads to a whole bunch of stuff that can be deleted around old AppDomain logic, and given our stance on this assembly, I decided to just leave things as they are.